### PR TITLE
Allow to set empty string for XmlPoke Value

### PR DESCRIPTION
**Pull request by JaynieBai**
Fixes [#5814](https://github.com/dotnet/msbuild/issues/5814)

### Context
Can't pass empty string to XmlPoke task.

### Changes Made
Remove the required attribute for Value and the exception when value is null.

### Testing
Unit test XmlPokeWithEmptyValue()

### CODE REVIEW

File: src/Tasks/XmlPoke.cs [hunk 1]
```diff
@@ -68,16 +69,10 @@ public string Query
...
         public ITaskItem Value
         {
-            get
-            {
-                ErrorUtilities.VerifyThrowArgumentNull(_value, nameof(Value));
-                return _value;
-            }
-
+            get => _value;
             set => _value = value;
```
Comment (JanKrivanek) [inline src/Tasks/XmlPoke.cs:76]: Do we need the backing field?

File: src/Tasks.UnitTests/XmlPoke_Tests.cs [hunk 2]
```diff
@@ -174,6 +169,23 @@ public void PokeMissingParams()
...
+            string projectContents = @"
+                <Project ToolsVersion='msbuilddefaulttoolsversion'>
+                <Target Name='Poke'>
+                    <XmlPoke Value='' Query='//class/variable/@Name' XmlInputPath='{0}'/>
+                </Target>
+                </Project>";
+            projectContents = string.Format(projectContents, xmlInputPath);
+
+            ObjectModelHelpers.BuildProjectExpectSuccess(projectContents);
+        }
```
Comment (JanKrivanek) [inline src/Tasks.UnitTests/XmlPoke_Tests.cs:187]: We should probably assert here that the target attribute/element is still present, but with empty value.

File: src/Tasks.UnitTests/XmlPoke_Tests.cs [hunk 3]
```diff
@@ -174,6 +169,23 @@ public void PokeMissingParams()
             }
         }
 
+        [Fact]
+        // https://github.com/dotnet/msbuild/issues/5814
+        public void XmlPokeWithEmptyValue()
+        {
+            string xmlInputPath;
+            Prepare(_xmlFileNoNs, out xmlInputPath);
+            string projectContents = @"
```
Comment (JanKrivanek) [inline src/Tasks.UnitTests/XmlPoke_Tests.cs:178]: No need for change here - just an information. From C# 11 you can use [raw string literals](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-11#raw-string-literals) - sou you wouldn't need to combine quotes here.

```C#
string projectContents = """
<Project ToolsVersion="msbuilddefaulttoolsversion">
                <Target Name="Poke">
                    <XmlPoke Value="" Query="//class/variable/@Name" XmlInputPath="{0}"/>
                </Target>
                </Project>""";
```

**Review by JanKrivanek — CHANGES_REQUESTED** [review]
Simple and effective!
I'd suggest to improve the test case to varify the behavior (resulting attribute is present, but empty)

**Review by JanKrivanek — APPROVED** [review]
Looks good!
Thank you for updating the test case!

**Review by Forgind — APPROVED** [review]
LGTM!

**Comment by JaynieBai** [comment]
@ghogen  This PR make making the Value property of XmlPoke task optional. please help update document https://github.com/MicrosoftDocs/visualstudio-docs/blob/main/docs/msbuild/xmlpoke-task.md

**Pull request merged**